packages/utilities/src/BaseComponent.ts

```diff
@@ -58,23 +58,20 @@ export class BaseComponent<TProps extends IBaseProps = {}, TState = {}> extends
     initializeDir();
 
     _makeAllSafe(this, BaseComponent.prototype, [
-      'componentWillMount',
       'componentDidMount',
       'shouldComponentUpdate',
-      'componentWillUpdate',
-      'componentWillReceiveProps',
+      'getSnapshotBeforeUpdate',
       'render',
       'componentDidUpdate',
       'componentWillUnmount'
     ]);
   }
 
   /**
-   * When the component will receive props, make sure the componentRef is updated.
+   * When the component receives props, make sure the componentRef is updated.
    */
-  // tslint:disable-next-line:no-any
-  public componentWillReceiveProps(newProps: Readonly<TProps>, newContext: any): void {
-    this._updateComponentRef(this.props, newProps);
+  public componentDidUpdate(prevProps: TProps, prevState: TState): void {
+    this._updateComponentRef(prevProps, this.props);
```

Comment:
Just thinking in writing here. Hmm... this is subtle. It's changing the ref wrapper prop from being updated before render to after render.

componentRef is basically a wrapper that captures the reference value when it updates, which is "sometime" after render, but guaranteed to be set before post-mount lifecycle methods are called. But this applies to DOM nodes and I don't think componentRef is actually ever set to a DOM node, so I think we're ok. It seems that componentRef is only ever set to the instance of the component, which wouldn't change before and after render.

Reply:
Thanks for the thoughts! That's basically what I wanted in asking so many people to look at this. :) 

More thinking in writing...is there any time during render that the parent component (which just passed in a new `componentRef` prop) might want to access that ref? Saying that makes me wonder, why would someone change the `componentRef` to begin with...since that seems pretty unlikely, it means **not many consumers should be impacted by this change.**

Also noticed that initial `componentRef` resolution happens **after render** in `componentDidMount`--which means updating after render should also be safe.

Reply:
I think we're being consistent with React behavior. No consumer should expect to be able to use a new instance of componentRef until AFTER render.